java-frontend/src/main/java/org/sonar/java/model/JInitializerBlockSymbol.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.model;
+
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.semantic.SymbolMetadata;
+import org.sonar.plugins.java.api.semantic.Type;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.MethodTree;
+
+public class JInitializerBlockSymbol implements Symbol.MethodSymbol {
+  private final TypeSymbol owner;
+  private final boolean isStatic;
+
+  public JInitializerBlockSymbol(TypeSymbol owner, boolean isStatic) {
+    this.owner = owner;
+    this.isStatic = isStatic;
+  }
+
+  @Override
+  public MethodTree declaration() {
```

Comment:
I'm a bit annoyed by this. I would like to have a link to its declaration, especially if it exists.
What about implementing _only_ the `Symbol` interface instead of `MethodSymbol`?

Reply:
I've thought about it, but left it as-is. If the class didn't implement `MethodSymbol`, then any code doing "if `isMethodSymbol()`, then cast to `MethodSymbol`" would break - and we do that a lot. The fix would be to make `isMethodSymbol()` return `false` and introduce `isInitializerBlockSymbol()` which would return `true`. That might be better, but would mean introducing a new method to the API.